src/runtime.js

```diff
@@ -545,12 +545,18 @@ function prepareAndRegisterServiceForDocShadowMode(global, extensions,
  * @param {function(!./service/ampdoc-impl.AmpDoc):!Object=} opt_factory
  */
 function registerServiceForDoc(ampdoc, name, opt_ctor, opt_factory) {
+  // TODO(kmh287, #9292): Refactor to remove opt_factory param once #9212 has been
```

Comment:
This looks great! Please also add `TODO(kmh287, #9292)` with the corresponding explanation of what will be done to `prepareAndRegisterServiceForDoc` and `prepareAndRegisterServiceForDocShadowMode`.